Code/Unsupervised_reconstruction/data/reconstruction_datasets.py

```diff
@@ -48,4 +53,57 @@ def trainingDataLoader(raw_samples:list[str],
 
 
 #-------------Sampling Datasets---------------------
+Raw_minibatch_type = tuple[tuple[
+                        tuple[Tensor], # tuple of c ByteTensors of shape (b, |x|)
+                        list[dict[ModernLanguages, Tensor]] # list of c dicts with the c cognates group
+                        ],
+                        tuple[int, int] # the coords of the minibatch in the global batch
+                        ]
 
+class __SamplingDataset(IterableDataset[Raw_minibatch_type]):
```

Comment:
Why the choice of an `IterableDataset`?

Reply:
The global dataset does not have to be entirely loaded to transform and process the sequence of mini-batches. Moreover, the cutting in mini-batches does not require any shuffling operation and can be carried out in a sequential order. The `IterableDataset` class enables to control the process flow in this manner with letting a huge dataset be loaded within, for instance by chaining several instances of it (see the usage of `ChainingDataset` in this script for an example).

I concede the better usage would be to just give the filename of a .pth for constructing a `__SamplingDataset` instance. If we note C the total number of cognate pairs and B the number of proposals to be processed per cognate pair, this file would contain here a sublist of c tensors with B proposals. Then, if the C tensors of B proposals had had to be splitted into sublists of c tensors of B proposals for not saturing the memory, a list of ~C/c files --- each containing a sublist --- would be able to be loaded in a chained dataset of `__SamplingDataset`s.  
But in our case we do not need to load several files and can use the whole generated list of C tensors in memory, so the IterableDataset is only used for loading the mini_batches approximately in the order that fits their coordinates (if several workers are deployed, multiprocessing may break a bit this order but it does not matter at all).

Reply:
IterableDatasets are perfect for this situation. And in our case, as you mentioned, there's no need to create files for each instance of __SamplingDataset, they can be stored directly in memory.